crates/astria-sequencer/src/state_ext.rs

```diff
@@ -243,3 +243,298 @@ pub(crate) trait StateWriteExt: StateWrite {
+                .await
+                .unwrap()
+                == storage_version_update
+        );
+    }
+
+    #[tokio::test]
+    async fn native_asset_denom() {
+        let storage = cnidarium::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        // doesn't exist at first
+        assert!(state.get_native_asset_denom().await.is_err());
+
+        // can write
+        let denom_orig = "denom_orig";
+        state.put_native_asset_denom(denom_orig);
+        assert!(state.get_native_asset_denom().await.unwrap() == denom_orig);
+
+        // can write new value
+        let denom_update = "denom_update";
+        state.put_native_asset_denom(denom_update);
+        assert!(state.get_native_asset_denom().await.unwrap() == denom_update);
+    }
+
+    #[tokio::test]
+    async fn block_fee_read_and_increase() {
+        let storage = cnidarium::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        // doesn't exist at first
+        let fee_balances_orig = state.get_block_fees().await.unwrap();
+        assert!(fee_balances_orig.is_empty());
+
+        // can write
+        let asset = astria_core::sequencer::v1::asset::Id::from_denom("asset_0");
+        let amount = 100u128;
+        state
+            .get_and_increase_block_fees(asset, amount)
+            .await
+            .unwrap();
+
+        // holds expected
+        let fee_balances_updated = state.get_block_fees().await.unwrap();
+        assert!(fee_balances_updated[0] == (asset, amount));
+    }
+
+    #[tokio::test]
+    async fn block_fee_read_and_increase_can_delete() {
+        let storage = cnidarium::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        // can write
+        let asset_first = astria_core::sequencer::v1::asset::Id::from_denom("asset_0");
+        let asset_second = astria_core::sequencer::v1::asset::Id::from_denom("asset_1");
+        let amount_first = 100u128;
+        let amount_second = 200u128;
+
+        state
+            .get_and_increase_block_fees(asset_first, amount_first)
+            .await
+            .unwrap();
+        state
+            .get_and_increase_block_fees(asset_second, amount_second)
+            .await
+            .unwrap();
+        // holds expected
+        let fee_balances = state.get_block_fees().await.unwrap();
+        for val in fee_balances {
+            assert!(val == (asset_first, amount_first) || val == (asset_second, amount_second));
+        }
```

Comment:
i think you can just construct the vec and compare directly: `vec![(asset_first, amount_first), (asset_second, amount_second)]`